### PR TITLE
Refactor styles & add widget/golden tests

### PR DESCRIPTION
**Pull request by Pante**
This PR:
* Refactor styles into the same file
* Introduce widget tests
* Introduce golden tests

### CODE REVIEW

File: forui/lib/src/widgets/card.dart [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+/// A card, typically with a title and subtitle.
+final class FCard extends StatelessWidget {
+  /// The child.
+  final Widget child;
+
+  /// The style.
+  final FCardStyle? style;
+
+  /// Creates a [FCard].
```
Comment (kawaijoe) [inline forui/lib/src/widgets/card.dart:18]:
```suggestion
  /// Creates a [FCard] with a tile and subtitle.
```

File: forui/test/src/theme/theme_test.dart [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      final updated = tester.widget<Text>(find.byKey(key)).data;
+
+      expect(initial, isNot(updated));
+    });
+
+    testWidgets('no ThemeData in ancestor', (tester) async {
+      await tester.pumpWidget(Directionality(
+        textDirection: TextDirection.ltr,
+        child: Builder(builder: (context) => Text('${context.theme == FThemes.zinc.dark}')),
+      ));
```
Comment (kawaijoe) [inline forui/test/src/theme/theme_test.dart:63]: Missing trailing comma.

File: forui/test/src/theme/theme_test.dart [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      expect(find.text('false'), findsOneWidget);
+    });
+
+    testWidgets('inherit TextDirection from parent', (tester) async {
+      await tester.pumpWidget(MaterialApp(
+        home: FTheme(
+          data: FThemes.zinc.dark,
+          child: const Text(''),
+        ),
+      ));
```
Comment (kawaijoe) [inline forui/test/src/theme/theme_test.dart:74]: Missing trailing comma.

**Review by kawaijoe — CHANGES_REQUESTED** [review]

File: forui/lib/src/widgets/box.dart
[the remark this answers is not in the document]
Comment (kawaijoe) [reply forui/lib/src/widgets/box.dart:1]: Edit: I think widgets with a single file can stay in the root directory. For example, `Card` definitely needs its own folder.

With that said, I think it is generally a good guideline to separate each class to it's own file.
https://stackoverflow.com/a/374274/2801180

TLDR; I think the previous structure is superior.

[the remark this answers is not in the document]
Comment (Pante) [reply forui/lib/src/widgets/box.dart:1]: I'm okay with moving the card/button files back to their own folders.

While I agree with the reasons outlined in the answer, I think that there's some context to it. The answer is to a question talking about several classes in a file which totals ~7k lines. In such cases, I definitely agree that it should be split. However, most of our files are 100+ lines long which IMO is quite short. Most of the other reasons are associated with really large files too which isn't really applicable to our case.

File: forui/test/src/box.dart
[the remark this answers is not in the document]
Comment (Pante) [reply forui/test/src/box.dart:1]: This is to align the widget nicely in the centre & add a background. This is to ensure we can see the border of the widget clearly & make sure we actually apply a background to the widget under test (Card previously forgot to set it's background color).

[the remark this answers is not in the document]
Comment (kawaijoe) [reply forui/test/src/box.dart:1]: I see. I think the naming could be improved, may I suggest `TestScaffold`?

File: forui/lib/src/widgets/box.dart
[the remark this answers is not in the document]
Comment (kawaijoe) [reply forui/lib/src/widgets/box.dart:1]: Fair enough
